Source/Request.swift

```diff
@@ -35,7 +35,16 @@ public class Request {
     // MARK: Properties
 
     /// The delegate for the underlying task.
-    public let delegate: TaskDelegate
+    public internal(set) var delegate: TaskDelegate {
+        get {
+            taskDelegateLock.lock() ; defer { taskDelegateLock.unlock() }
+            return taskDelegate
+        }
+        set {
+            taskDelegateLock.lock() ; defer { taskDelegateLock.unlock() }
+            taskDelegate = newValue
+        }
+    }
```

Comment:
This may be overkill to wrap the change here in a lock, but I figure it's better safe than sorry. The only time this can ever change is when a data task is converted to a download task. Previously we were changing the subdelegate and not actually updating the request. That then created a mismatch between what the `Request` delegate reported and what was actually called.